### PR TITLE
Add COVID-19 candidate banner to ‘Your application’ page

### PR DESCRIPTION
**Pull request by C-gyorfi**
## Context

We need to update candidate UI with the COVID 19 banner with the guidance for candidates

## Changes proposed in this pull request

This PR adds the banner to the ‘Your application’ page

#### After
image

## Guidance to review

The acceptance criteria say:
- only shows if no other banner is not activated
Not sure if I need to add any more logic to prevent this?

## Link to Trello card

https://trello.com/c/lBLa3fwp/1217-dev-covid-19-candidate-banner

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by davidgisbey** [comment]
In the card, it says only to show if no other banner is active. Not sure if that means it shouldn't show if a flash message is present or not? It's really not a big deal btw. Sorry if this is nit-picking.

image

**Comment by C-gyorfi** [comment]
> In the card, it says only to show if no other banner is active. Not sure if that means it shouldn't show if a flash message is present or not? It's really not a big deal btw. Sorry if this is nit-picking.
> 
> image

Good shout @davidgisbey! Is this an issue? @paulrobertlloyd

File: spec/system/candidate_interface/candidate_viewing_application_spec.rb [hunk 1]
```diff
@@ -7,6 +7,11 @@
     given_i_am_signed_in
     when_i_visit_the_site
     then_i_should_see_that_i_have_made_no_choices
+    and_i_should_not_see_the_covid19_banner
+
+    when_covid19_feature_flag_is_active
+    and_i_visit_the_site
```
Comment (chubberlisk) [inline spec/system/candidate_interface/candidate_viewing_application_spec.rb]: Nitpick, feel free to ignore!

```suggestion
    given_covid19_feature_flag_is_active
    when_i_visit_the_site
```

**Comment by C-gyorfi** [comment]
@chubberlisk @davidgisbey Updated in 33da84e to meet the `only shows if no other banner is not activated` criteria

**Comment by paulrobertlloyd** [comment]
@C-gyorfi Yup, just to confirm, only one banner should be shown in any given state. If no other banner/flash message is shown, show the Covid-19 banner.

File: app/views/candidate_interface/application_form/show.html.erb [hunk 2]
```diff
@@ -1,5 +1,14 @@
 <% content_for :title, @application_form.amendable? ? t('page_titles.edit_application_form') : t('page_titles.application_form') %>
 
+<% if FeatureFlag.active?('covid_19') && flash[:success].nil? %>
```
Comment (chubberlisk) [inline app/views/candidate_interface/application_form/show.html.erb]: It's possible to have different types of `flash`: `flash[:success]`, `flash[:warning]` and `flash[:info]` - see `app/components/flash_message_component.html.erb`. So I think this should be:

```suggestion
<% if FeatureFlag.active?('covid_19') && flash.empty? %>
```

Does that make sense?

Comment (C-gyorfi) [reply app/views/candidate_interface/application_form/show.html.erb]: Yep that's right!

**Pull request merged**